src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -194,43 +194,47 @@ impl Display for &'_ PartialGraphManagedBarrierState {
 enum InflightActorStatus {
     /// The actor is just spawned and not issued any barrier yet
     NotStarted,
-    /// The actor has been issued some barriers, and not issued any stop barrier yet
+    /// The actor has been issued some barriers, but has not collected the first barrier
+    IssuedFirst(Vec<Barrier>),
+    /// The actor has been issued some barriers, and has collected the first barrier
     Running(u64),
-    /// The actor has been issued a stop barrier
-    Stopping(u64),
 }
 
 impl InflightActorStatus {
-    pub(super) fn is_stopping(&self) -> bool {
-        matches!(self, InflightActorStatus::Stopping(_))
-    }
-
     fn max_issued_epoch(&self) -> Option<u64> {
         match self {
             InflightActorStatus::NotStarted => None,
-            InflightActorStatus::Running(epoch) | InflightActorStatus::Stopping(epoch) => {
-                Some(*epoch)
+            InflightActorStatus::Running(epoch) => Some(*epoch),
+            InflightActorStatus::IssuedFirst(issued_barriers) => {
+                Some(issued_barriers.last().expect("non-empty").epoch.prev)
             }
         }
     }
 }
 
 pub(crate) struct InflightActorState {
+    actor_id: ActorId,
     pending_subscribers: BTreeMap<u64, Vec<mpsc::UnboundedSender<SubscribeMutationItem>>>,
+    barrier_senders: Vec<mpsc::UnboundedSender<Barrier>>,
```

Comment:
In which case will we get multiple barrier senders for the same actor?

Reply:
In the current main we support having multiple senders from a single actor, but I have no context on the original design. But in some future PRs we can leverage this to replace the current barrier mutation subscriber and unify to use this barrier senders.